apps/website/src/pages/events/fall-carnival-2023/tickets/[userName].tsx

```diff
@@ -0,0 +1,154 @@
+import type { GetStaticPaths, GetStaticProps, NextPage } from "next";
+import { useRouter } from "next/router";
+import Image from "next/image";
+import NextLink from "next/link";
+
+import { env } from "@/env/index.mjs";
+
+import { eventId, ticketConfig } from "@/data/events/fall-carnival-2023";
+
+import leafLeftImage1 from "@/assets/floral/leaf-left-1-fall.png";
+import leafLeftImage2 from "@/assets/floral/leaf-left-2.png";
+import leafRightImage2 from "@/assets/floral/leaf-right-2-fall.png";
+
+import { trpc } from "@/utils/trpc";
+import { getVirtualTicketImageUrl } from "@/utils/virtual-tickets";
+
+import Section from "@/components/content/Section";
+import Meta from "@/components/content/Meta";
+
+import { Wiggle } from "@/components/events/virtual-ticket/Wiggle";
+
+import { IntroSection } from "@/components/events/fall-carnival-2023/IntroSection";
+import { Activities } from "@/components/events/fall-carnival-2023/Activities";
+
+export const getStaticPaths = (async () => {
+  return {
+    paths: [],
+    fallback: "blocking",
+  };
+}) satisfies GetStaticPaths;
+
+export const getStaticProps = (async () => ({
+  props: {},
+})) satisfies GetStaticProps;
+
+const TicketPage: NextPage = () => {
+  const router = useRouter();
+
+  const userName = router.query.userName
+    ? String(router.query.userName).trim()
+    : undefined;
+
+  const ticket = trpc.virtualTickets.getTicket.useQuery(
+    { eventId, userName: userName as string },
+    { enabled: !!userName, retry: 2 },
+  );
+
+  const showTicket = userName && ticket.data;
+
+  const ogImage =
+    userName &&
```

Comment:
Instead of repeatedly checking if the username is set, maybe we can use `invariant` to check instead.